Core/Object Arts/Dolphin/Base/DateAndTimeTest.cls

```diff
@@ -486,6 +486,36 @@ testStbConvertFrom
 				actual := Object fromLiteralStoreArray: stl.
 				self assert: actual equals: expected]!
 
+testStbConvertFrom0
+	"Test upgrade from version 0.
+	The stored instances I have do not have a Duration instance for seconds (JGF)"
+
+	"Create a ByteArray from 96fc6fa776aff7f8e76246917a28cda0195e27be (24 April 2019)
+	
+	| ts date time dat |
+	ts := TimeStamp current.
+	date := ts date.
+	time := ts time.
+	time seconds class == Fraction ifFalse: [self error: 'wrong class!!'].
+	dat := DateAndTime
+		year: date year 
+		month: date month 
+		day: date dayOfMonth 
+		hour: time hour 
+		minute: time minute 
+		second: time second.
+	dat binaryStoreBytes.
+	 "
+
+	| bytes dat string |
+	bytes := #[33 83 84 66 32 52 32 6 3 11 0 68 97 116 101 65 110 100 84 
+		105 109 101 131 127 22 0 6 2 8 0 70 114 97 99 116 105 111 110 183 
+		217 116 4 209 7 0 0 33 59 255 255].	"Obtained from the code above"
+	dat := DateAndTime fromBinaryStoreBytes: bytes.
+	string := dat printString.
+	self assert: string = ' 2019-06-07T10:23:03.387-07:00'.	"What is with the leading space?"
```

Comment:
ANSI NCITS 319-1998, <DateAndTime> protocol definition, section 5.8.1.24, Message refinement: printString
```
...
The result will be formatted as
 -YYYY-MM-DDThh:mm:ss.s+ZZ:zz.z where
 -     is the <Character> $- if the year is less than 0 otherwise it is the <Character> that is 
        returned from the message #space sent to the global Character,
...
```
It means that the result is always of the same width (assuming the same precision is used for fractional seconds), but I find it a strange choice too. You can omit it by dropping down a level to #printOn:withLeadingSpace:, or use `#printStringFormat:` to have complete control over the format.